### PR TITLE
Modify the format of periodic upload data

### PR DESCRIPTION
**Pull request by gc87**

### CODE REVIEW

File: include/json/neu_json_mqtt.h [hunk 1]
```diff
@@ -43,10 +43,18 @@ typedef struct {
     char *command;
 } neu_json_mqtt_t;
 
+typedef struct {
+    char *   config_name;
+    char *   node_name;
+    uint32_t node_id;
+    uint64_t timestamp;
+} neu_json_mqtt_periodic_t;
+
```
Comment (fengzeroz) [inline include/json/neu_json_mqtt.h:52]: put it to read/write json, eKuiper will also use it

File: plugins/mqtt/command/read_write.c [hunk 2]
```diff
@@ -20,11 +20,20 @@
...
 
-int command_read_once_request(neu_plugin_t *plugin, neu_json_mqtt_t *mqtt,
-                              neu_json_read_req_t *req, uint32_t req_id)
+static uint64_t current_time()
+{
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    uint64_t ms = tv.tv_sec;
+    return ms * 1000 + tv.tv_usec / 1000;
+}
```
Comment (fengzeroz) [inline plugins/mqtt/command/read_write.c:33]: when the number of messages is large, calling gettimeofday too many times is inefficient

File: plugins/mqtt/command/read_write.c [hunk 3]
```diff
@@ -226,15 +235,61 @@ char *command_read_once_response(neu_plugin_t *   plugin,
     return json_str;
 }
 
-char *command_read_cycle_response(neu_plugin_t *  plugin,
-                                  neu_data_val_t *resp_val)
+static bool node_match(const void *key, const void *item)
+{
+    neu_node_id_t    id   = *(neu_node_id_t *) key;
+    neu_node_info_t *info = (neu_node_info_t *) item;
+    return (id == info->node_id) ? true : false;
```
Comment (fengzeroz) [inline plugins/mqtt/command/read_write.c:242]: id==info->node_id

File: plugins/mqtt/command/read_write.c [hunk 4]
```diff
@@ -226,15 +235,61 @@ char *command_read_once_response(neu_plugin_t *   plugin,
...
-    neu_uuid_v4_gen(uuid4_str);
-    parse_header.uuid    = uuid4_str;
-    parse_header.command = "";
-    return command_read_once_response(plugin, &parse_header, resp_val);
+    neu_fixed_array_t *array = NULL;
+    neu_dvalue_get_ref_array(resp_val, &array);
+    if (NULL == array || 0 >= array->length) {
+        log_error("Get array error");
+        return NULL;
+    }
```
Comment (fengzeroz) [inline plugins/mqtt/command/read_write.c:254]: problems caused by coding use assert

File: include/json/neu_json_mqtt.h (hunk 1 above)
Comment (gc87) [reply include/json/neu_json_mqtt.h]: Done.

File: src/core/message.h [hunk 5]
```diff
@@ -174,6 +174,7 @@ typedef struct grp_config_changed_event {
 /* MSG_DATA_NEURON_TRANS_DATA */
 typedef struct neuron_trans_data {
     neu_taggrp_config_t *grp_config;
+    adapter_id_t         sender_id; // adapter_id of sender
```
Comment (fengzeroz) [inline src/core/message.h:177]: add node_name field, to avoid query nodes vector

**Review by fengzeroz — APPROVED** [review]

**Pull request merged**